llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -2511,6 +2511,13 @@ Instruction *InstCombinerImpl::visitCallInst(CallInst &CI) {
   }
   case Intrinsic::fabs: {
     Value *Cond, *TVal, *FVal;
+    Value* Arg = II->getArgOperand(0);
+    Value* X;
+    // fabs (-X) --> fabs (X)
+    if (match(Arg, m_FNeg(m_Value(X))))
```

Comment:
The comment https://github.com/llvm/llvm-project/pull/94183#discussion_r1624729580 is not resolved.